packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/utils.dart

```diff
@@ -419,3 +442,64 @@ class AnimatedFlexLayoutProperties
     );
   }
 }
+
+/// Enum object to represent which side of the widget is overflowing.
+///
+/// See also:
+/// * [OverflowIndicatorPainter]
+enum OverflowSide {
+  right,
+  bottom,
+}
+
+OverflowSide overflowSide(LayoutProperties properties) {
+  if (properties.overflowWidth) return OverflowSide.right;
+  if (properties.overflowHeight) return OverflowSide.bottom;
+  return null;
+}
+
+/// CustomPainter for drawing [DebugOverflowIndicatorMixin] patterned background.
```

Comment:
I'd be fine with also adding a separate
overflow_painter.dart file for this class rather than leaving it in utils.